src/uu/pmap/src/smaps_format_parser.rs

```diff
@@ -0,0 +1,461 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use crate::maps_format_parser::{parse_map_line, MapLine};
+use std::io::{Error, ErrorKind};
+
+// Represents a parsed single entry from /proc/<PID>/smaps for the extended formats.
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct SmapEntry {
+    pub map_line: MapLine,
+    pub size_in_kb: u64,
+    pub kernel_page_size_in_kb: u64,
+    pub mmu_page_size_in_kb: u64,
+    pub rss_in_kb: u64,
+    pub pss_in_kb: u64,
+    pub shared_clean_in_kb: u64,
+    pub shared_dirty_in_kb: u64,
+    pub private_clean_in_kb: u64,
+    pub private_dirty_in_kb: u64,
+    pub referenced_in_kb: u64,
+    pub anonymous_in_kb: u64,
+    pub lazy_free_in_kb: u64,
+    pub anon_huge_pages_in_kb: u64,
+    pub shmem_pmd_mapped_in_kb: u64,
+    pub file_pmd_mapped_in_kb: u64,
+    pub shared_hugetlb_in_kb: u64,
+    pub private_hugetlb_in_kb: u64,
+    pub swap_in_kb: u64,
+    pub swap_pss_in_kb: u64,
+    pub locked_in_kb: u64,
+    pub thp_eligible: u64,
+    pub vmflags: String,
+}
+
+// Parses entries from /proc/<PID>/smaps. See
+// https://www.kernel.org/doc/html/latest/filesystems/proc.html for details about the expected
```

Comment:
maybe move this link to the struct above would be better?

Reply:
`proc_pid_smaps(5)` also in manpage

https://manpages.ubuntu.com/manpages/oracular/man5/proc_pid_smaps.5.html